### PR TITLE
Add authentication feature to comments page.

### PR DESCRIPTION
**Pull request by asam01**
Changes:
- LoginServlet.java - connects to authentication API, sends a login prompt if user has not logged in
- LoginStatusServlet.java - verifies login status (logged in or out)
- forum.html - add message to tell if user is logged in or out
- script.js - hides comments if not logged in, updates login status

https://screenshot.googleplex.com/CYow65VzBJn.png : before login
https://screenshot.googleplex.com/e7OQwLO6dfm.png : directed to login page
https://screenshot.googleplex.com/5V4QcFapwLU.png : after clicking log in
https://screenshot.googleplex.com/SF5AYdbAWoy.png : after clicking log out

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    
+    UserService userService = UserServiceFactory.getUserService();
+    response.setContentType("text/html;");
+
+    if (userService.isUserLoggedIn()) {
+      
+      response.getWriter().println("<p>Hello " + userService.getCurrentUser().getEmail() + "!</p>");
+      String logoutUrl = userService.createLogoutURL("/forum.html");
+      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a></p>");
+
```
Comment (hairong1007) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:47]: nit: Remove the blank line.

Comment (asam01) [reply portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: will fix

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -135,4 +135,32 @@ function createMap() {
...
+}
+
+async function getLoginStatus() {
+
+  const response = await fetch('/login-status');
+  const status = await response.json();
+ 
+  // user is logged in, allow them to view comments
+  if(status === "true") {
+    document.getElementById('comment-container').style.visibility = 'visible';
```
Comment (hairong1007) [inline portfolio/src/main/webapp/script.js:147]: Sometimes the ternary operator (condition ? exprIfTrue : exprIfFalse) can replace if statement.

Comment (asam01) [reply portfolio/src/main/webapp/script.js:147]: will keep in mind for next version!

**Pull request merged**